plugin/src/main/java/org/autorefactor/refactoring/rules/CollectionRefactoring.java

```diff
@@ -398,28 +401,46 @@ private boolean isZero(Expression expr) {
-            return maybeReplaceSetContains(miContains, node.getElseStatement(), miContains, true);
+            return maybeReplaceSetContains(node, node.getExpression(), node.getElseStatement(), node.getThenStatement(),
+                    true);
         }
     }
 
-    private boolean maybeReplaceSetContains(
-            MethodInvocation miContains, Statement stmt, Expression toReplace, boolean negate) {
-        if (isMethod(miContains, "java.util.Set", "contains", "java.lang.Object")) {
-            Statement firstStmt = getAsList(stmt, 0);
-            MethodInvocation miAdd = asExpression(firstStmt, MethodInvocation.class);
+    private boolean maybeReplaceSetContains(final IfStatement nodeToReplace, final Expression ifExpression,
+            final Statement statement,
+            final Statement oppositeStatement, final boolean negate) {
+        return maybeReplaceSetContains(nodeToReplace, ifExpression, statement, oppositeStatement, negate, "add")
+                &&  maybeReplaceSetContains(nodeToReplace, ifExpression, oppositeStatement, statement, !negate,
+                    "remove");
+    }
+
+    private boolean maybeReplaceSetContains(final IfStatement nodeToReplace, final Expression ifExpression,
+            final Statement statement,
+            final Statement oppositeStatement, final boolean negate, final String methodName) {
+        final MethodInvocation miContains = as(ifExpression, MethodInvocation.class);
+        if (isMethod(miContains, "java.util.Set", "contains", "java.lang.Object") && !asList(statement).isEmpty()) {
+            final Statement firstStmt = getAsList(statement, 0);
+            final MethodInvocation miAddOrRemove = asExpression(firstStmt, MethodInvocation.class);
             final ASTMatcher astMatcher = new ASTMatcher();
-            if (isMethod(miAdd, "java.util.Set", "add", "java.lang.Object")
-                    && match(astMatcher, miContains.getExpression(), miAdd.getExpression())
-                    && match(astMatcher, arg0(miContains), arg0(miAdd))) {
-                ASTBuilder b = this.ctx.getASTBuilder();
-                Refactorings r = this.ctx.getRefactorings();
-                r.replace(toReplace, negate ? b.negate(miAdd, COPY.MOVE) : b.move(miAdd));
-                r.remove(firstStmt);
-                return DO_NOT_VISIT_SUBTREE;
+            if (isMethod(miAddOrRemove, "java.util.Set", methodName, "java.lang.Object")
+                    && match(astMatcher, miContains.getExpression(), miAddOrRemove.getExpression())
+                    && match(astMatcher, arg0(miContains), arg0(miAddOrRemove))) {
+                final ASTBuilder b = this.ctx.getASTBuilder();
+                final Refactorings r = this.ctx.getRefactorings();
+
+                if (asList(statement).size() == 1 && asList(oppositeStatement).isEmpty()) {
+                    // Only one statement: add() or remove()
+                    r.replace(nodeToReplace, b.copy(firstStmt));
```

Comment:
(no need to act on this, I fixed it in my local copy)

Try to use `ASTBuilder.move()` wherever possible. 
It will preserve comments as much as possible, while `ASTBuilder.copy()` will only copy the AST, but not the comments.
For some obscure reasons we cannot always use `ASTBuilder.move()` and need to resort using `ASTBuilder.copy()` or worse, `ASTBuilder.copySubtree()`, but the latter is fairly rare.